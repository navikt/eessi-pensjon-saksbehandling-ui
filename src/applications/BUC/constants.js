export const BUC_MODE = 'BUC/MODE'
