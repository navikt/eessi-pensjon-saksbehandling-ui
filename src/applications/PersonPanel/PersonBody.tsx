import { PersonIcon, HouseIcon, CalendarIcon, GlobeIcon, PersonRectangleIcon, HeartIcon } from '@navikt/aksel-icons'
import PostalCodes from 'src/components/PostalCodes/PostalCodes'
import { AllowedLocaleStringPropType } from 'src/declarations/app.pt'
import { PersonAvdod, PersonAvdods } from 'src/declarations/person.d'
import CountryData from '@navikt/land-verktoy'
import _ from 'lodash'
import moment from 'moment'
import { BodyLong, Box, Detail, HStack } from '@navikt/ds-react'
import PT from 'prop-types'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'
import { HorizontalLineSeparator, HorizontalSeparatorSpan } from 'src/components/StyledComponents'
import PopoverCustomized from "src/components/Tooltip/PopoverCustomized";

const Element = styled.div`
  display: flex;
  align-items: baseline;
`
const MarginRow = styled(HStack)`
  margin: 1.5rem;
  display: flex;
`
const MarginColumn = styled.div`
  display: flex;
  align-items: flex-start;
  margin-bottom: 0.5rem;
  flex: ${(props: any) => props.flex || 1};
  margin-left: 0.5rem;
  margin-right: 0.5rem;
`
export const PersonBodyDiv = styled.div``

export interface PersonBodyProps {
  locale: string
  person: any
  personAvdods: PersonAvdods | undefined
}

const PersonBody: React.FC<PersonBodyProps> = ({
  locale,
  person,
  personAvdods
}: PersonBodyProps): JSX.Element | null => {
  const { t } = useTranslation()

  const renderEntity = (label: string, value: any): JSX.Element => {
    let _value
    if (!value) {
      _value = [t('ui:notRegistered')]
    } else {
      _value = typeof value === 'string' ? [value] : value
    }
    if (_.isEmpty(_value)) {
      _value = [t('ui:notRegistered')]
    }
    return (
      <Element
        id={'w-overview-PersonBody--element-' + label.replace('ui:', '')}
        data-testid={'w-overview-PersonBody--element-' + label.replace('ui:', '')}
      >
        <Detail>
          <strong>
            {t(label)}
          </strong>:
        </Detail>
        <Box paddingInline="4 0">
          <div className='navds-body-long'>
            {_value.map((val: any) => val)}
          </div>
        </Box>
      </Element>
    )
  }

  const addAddressLine = (
    address: Array<JSX.Element | string>,
    value: string | null,
    label: string,
    separator: JSX.Element
  ): Array<JSX.Element | string> => {
    if (value) {
      address.push((
        <PopoverCustomized label={<span>{label}</span>} key={"tooltip-" + + label + value}>
          <span key={value}>{value}</span>
        </PopoverCustomized>
      ))
      address.push(separator)
    }
    return address
  }

  const getCountry = (value: string): string | null => {
    const nationality = CountryData.getCountryInstance(locale).findByValue3(value)
    return nationality ? nationality.label : null
  }

  const getDates = (person: any, category: string): string => {
    let beginDate = _.get(person, category + '.gyldigFraOgMed')
    if (beginDate) {
      beginDate = moment(beginDate).format('DD.MM.YYYY')
    }
    let endDate = ''
    if (_.get(person, category + '.gyldigTilOgMed')) {
      endDate = _.get(person, category + '.gyldigTilOgMed')
      if (endDate) {
        endDate = moment(endDate).format('DD.MM.YYYY')
      }
    }
    return beginDate + ' - ' + endDate
  }

  const getAddress = (person: any, category: string): Array<JSX.Element | string> => {
    if (!_.get(person, category + '.vegadresse')) {
      return []
    }
    let adresse: Array<JSX.Element | string> = []
    const husnummer = _.get(person, category + '.vegadresse.husnummer')
    const husbokstav = _.get(person, category + '.vegadresse.husbokstav')
    adresse = addAddressLine(adresse, getDates(person, category), t('ui:fram-og-til'), <br key={0} />)
    adresse = addAddressLine(adresse, _.get(person, category + '.vegadresse.adressenavn'), t('ui:adressenavn'),
      husnummer || husbokstav ? <HorizontalSeparatorSpan key={1} /> : <br key={1} />)
    adresse = addAddressLine(adresse, husnummer, t('ui:husnummer'), husbokstav ? <HorizontalSeparatorSpan key={2} /> : <br key={2} />)
    adresse = addAddressLine(adresse, husbokstav, t('ui:husbokstav'), <br key={3} />)
    const zipCode = _.get(person, category + '.vegadresse.postnummer')
    if (zipCode) {
      adresse = addAddressLine(adresse, zipCode, t('ui:poststed'), <HorizontalSeparatorSpan key={4} />)
      adresse = addAddressLine(adresse, PostalCodes.get(zipCode), t('ui:city'), <br key={5} />)
    }
    return adresse
  }

  if (!person) {
    return null
  }

  let birthDateString
  let deathDateString
  let nationality: Array<string> = []
  let maritalStatus: Array<string> = []
  const bostedsadresse: Array<JSX.Element | string> = getAddress(person, 'bostedsadresse')
  const oppholdsadresse: Array<JSX.Element | string> = getAddress(person, 'oppholdsadresse')

  if (_.get(person, 'foedsel.foedselsdato')) {
    birthDateString = moment(person.foedsel.foedselsdato).format('DD.MM.YYYY')
  }
  if (_.get(person, 'doedsfall.doedsdato')) {
    deathDateString = moment(person.doedsfall.doedsdato).format('DD.MM.YYYY')
  }
  if (!_.isEmpty(person.statsborgerskap)) {
    nationality = person.statsborgerskap.map((l: any) => {
      let label = getCountry(l.land)
      if (l.gyldigFraOgMed) {
        label += ' (' + moment(l.gyldigFraOgMed).format('DD.MM.YYYY')
      }
      if (l.gyldigTilOgMed) {
        label += ' - ' + moment(l.gyldigTilOgMed).format('DD.MM.YYYY')
      }
      return label + ')'
    })
  }

  if (!_.isEmpty(person.sivilstand)) {
    maritalStatus = person.sivilstand.map((s: any) => {
      let type = s.type.toLowerCase()
      type = type.charAt(0).toUpperCase() + type.slice(1)
      let dateString = ''
      if (s.gyldigFraOgMed) {
        dateString = moment(s.gyldigFraOgMed).format('DD.MM.YYYY')
      }
      if (s.gyldigTilOgMed) {
        dateString += ' - ' + moment(s.gyldigTilOgMed).format('DD.MM.YYYY')
      }
      const label = t(type) +
        (type !== 'Null' && type !== 'Ugift' && dateString
          ? ' (' + dateString + ')'
          : '')
      return label
    })
  }

  return (
    <PersonBodyDiv data-testid="person-body-div">
      <MarginRow key="personBody-marginrow-1">
        <MarginColumn key="personBody-marginrow-1-margincolumn-1">
          <HStack gap="4">
            <HouseIcon title={t('ui:bostedsadresse')} fontSize="1.5rem" />
            {bostedsadresse ? renderEntity('ui:bostedsadresse', bostedsadresse) : null}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-1-margincolumn-2">
          <HStack gap="4">
            <CalendarIcon title={t('ui:birthdate')} fontSize="1.5rem" />
            {renderEntity('ui:birthdate', birthDateString)}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-1-margincolumn-3">
          <HStack gap="4">
            <PersonRectangleIcon title={t('ui:nationality')} fontSize="1.5rem"/>
            {renderEntity('ui:nationality', nationality)}
          </HStack>
        </MarginColumn>
      </MarginRow>
      <HorizontalLineSeparator />
      <MarginRow key="personBody-marginrow-2">
        <MarginColumn key="personBody-marginrow-2-margincolumn-1">
          <HStack gap="4">
            <GlobeIcon title={t('ui:oppholdsadresse')} fontSize="1.5rem" />
            {renderEntity('ui:oppholdsadresse', oppholdsadresse)}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-2-margincolumn-2">
          <HStack gap="4">
            <HeartIcon title={t('ui:marital-status')} fontSize="1.5rem"/>
            {renderEntity('ui:marital-status', maritalStatus)}
          </HStack>
        </MarginColumn>
        <MarginColumn key="personBody-marginrow-2-margincolumn-3"/>
      </MarginRow>
      {((personAvdods && personAvdods.length > 0) || deathDateString) && (
        <>
          <HorizontalLineSeparator />
          <MarginRow key="personBody-marginrow-3">

            <MarginColumn key="personBody-marginrow-3-margincolumn-1">
              <PersonIcon fontSize="1.5rem" />
              <Box paddingInline="4 0">
                <Detail size='small'>
                  <strong>{t('ui:deceased')}</strong>:
                </Detail>
               </Box>
              <div>
                {personAvdods && personAvdods.length > 0
                  ? personAvdods.map((avdod: PersonAvdod) => (
                    <Element
                      key={avdod.fnr}
                      id='w-overview-PersonBody--element-deceased'
                    >
                      <Box paddingInline="4 0">
                        <BodyLong>
                          {avdod?.fornavn +
                            (avdod?.mellomnavn ? ' ' + avdod?.mellomnavn : '') +
                            (avdod?.etternavn ? ' ' + avdod?.etternavn : '') +
                            ' - ' + avdod.fnr}
                          {avdod.relasjon ? ' (' + t('buc:relasjon-' + avdod?.relasjon) + ')' : ''}
                          {avdod?.doedsDato ? ' (DÃ¸dsdato: ' + avdod.doedsDato + ')' : ''}

                        </BodyLong>
                      </Box>
                    </Element>
                  ))
                  : (
                    <Element
                        key='noAvdod'
                        id='w-overview-PersonBody--element-deceased'
                      >
                        <Box paddingInline="4 0">
                          <BodyLong>
                            {t('buc:form-noAvdod')}
                          </BodyLong>
                        </Box>
                      </Element>
                      )}
              </div>
            </MarginColumn>
            {deathDateString
              ? (
                <MarginColumn key="personBody-marginrow-3-margincolumn-2">
                  <HStack gap="4">
                    <CalendarIcon fontSize="1.5rem" />
                    {renderEntity('ui:deathdate', deathDateString)}
                  </HStack>
                </MarginColumn>
                )
              : (
                <MarginColumn key="personBody-marginrow-3-margincolumn-2"/>
                )}
          </MarginRow>
        </>
      )}
    </PersonBodyDiv>
  )
}

PersonBody.propTypes = {
  locale: AllowedLocaleStringPropType.isRequired,
  person: PT.object
}

export default PersonBody
