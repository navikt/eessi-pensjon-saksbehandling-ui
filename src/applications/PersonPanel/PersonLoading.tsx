import LoadingImage from 'src/components/Loading/LoadingImage'
import LoadingText from 'src/components/Loading/LoadingText'
import { HorizontalSeparatorDiv } from '@navikt/hoykontrast'
import styled from 'styled-components'

const Title = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
`
const PlaceholderText = styled(LoadingText)`
   height: 40px;
   min-height: 40px;
   width: 100%;
`
const PersonLoading = () => {
  return (
    <Title>
      <LoadingImage />
      <HorizontalSeparatorDiv />
      <PlaceholderText />
    </Title>
  )
}

export default PersonLoading
