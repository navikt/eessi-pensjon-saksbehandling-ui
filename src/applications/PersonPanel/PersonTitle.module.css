.dead {
  filter: grayscale(100%)
}
