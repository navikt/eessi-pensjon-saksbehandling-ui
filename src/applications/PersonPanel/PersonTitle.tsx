import kvinne from 'src/assets/icons/icon-kvinne.png'
import mann from 'src/assets/icons/icon-mann.png'
import ukjent from 'src/assets/icons/icon-ukjent.png'
import classNames from 'classnames'
import { PersonPDL } from 'src/declarations/person'
import _ from 'lodash'
import moment from 'moment'
import {Heading, HStack, Link} from '@navikt/ds-react'
import {getFnr, getNPID} from 'src/applications/BUC/components/BUCUtils/BUCUtils'
import LoadingImage from "src/components/Loading/LoadingImage";
import {copyToClipboard} from "src/actions/app";
import {useDispatch} from "react-redux";
import {useState} from "react";
import {useTranslation} from "react-i18next";
import {FilesFillIcon, FilesIcon} from "@navikt/aksel-icons";
import styles from "src/assets/css/common.module.css";
import titleStyles from "./PersonTitle.module.css"

export interface PersonTitleProps {
  gettingPersonInfo: boolean
  person?: PersonPDL
}

const PersonTitle: React.FC<PersonTitleProps> = ({
  gettingPersonInfo, person
}: PersonTitleProps): JSX.Element => {
  const [hover, isHover] = useState<boolean>(false);
  const dispatch = useDispatch()
  const { t } = useTranslation()

  let birthDate: Date | undefined
  let deathDate: Date | undefined

  let age: number = 0

  if (!person || gettingPersonInfo) {
    return (
      <LoadingImage />
    )
  }

  if (_.get(person, 'foedselsdato.foedselsdato')) {
    birthDate = moment(person.foedselsdato.foedselsdato)?.toDate()
  }
  if (_.get(person, 'doedsfall.doedsdato')) {
    deathDate = moment(person.doedsfall?.doedsdato)?.toDate()
  }

  const endDate: Date = deathDate || new Date()
  if (birthDate && endDate) {
    age = moment(endDate).diff(moment(birthDate), 'years')
  }

  let kind: string = 'nav-unknown-icon'
  let src = ukjent

  if (person.kjoenn?.kjoenn === 'KVINNE') {
    kind = 'nav-woman-icon'
    src = kvinne
  } else if (person.kjoenn?.kjoenn === 'MANN') {
    kind = 'nav-man-icon'
    src = mann
  }

  const pid : string | undefined = getFnr(person) ? getFnr(person) : getNPID(person)

  return (
    <HStack gap="4" align="center">
      <img
        alt={kind}
        className={classNames({ [titleStyles.dead]: !_.isNil(deathDate) })}
        data-testid='w-persontitle--img'
        src={src}
        width={40}
      />
      <Heading size='medium'>
        {person.navn?.sammensattNavn} ({age}) - {pid}
        <Link
          onMouseEnter={() => isHover(true)}
          onMouseLeave={() => isHover(false)}
          title={t('ui:person-kopier-pid')} onClick={(e: any) => {
          e.preventDefault()
          e.stopPropagation()
          pid && dispatch(copyToClipboard(pid))
        }}
        >
          {hover ?
            <FilesFillIcon className={styles.copyFilledWithMargin} fontSize="1.5rem"/> :
            <FilesIcon className={styles.copyWithMargin} fontSize="1.5rem"/>
          }
        </Link>
      </Heading>
    </HStack>
  )
}

export default PersonTitle
