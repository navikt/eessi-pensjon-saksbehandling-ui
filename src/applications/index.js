import SelectPDF from './PDF_/SelectPDF/SelectPDF'
import EditPDF from './PDF_/EditPDF/EditPDF'
import GeneratePDF from './PDF_/GeneratePDF/GeneratePDF'

import Case from './Case/Case'
import PSelv from './PSelv/PSelv'
import PInfo from './PInfo/PInfo'
import PInfoSaksbehandler from './PInfo/PInfoSaksbehandler'
import P4000 from './P4000/P4000'
import NewFeatures from './NewFeatures/NewFeatures'
import BUC from './BUC/BUC'

export {
  SelectPDF,
  EditPDF,
  GeneratePDF,
  Case,
  PSelv,
  PInfoSaksbehandler,
  PInfo,
  P4000,
  NewFeatures,
  BUC
}
