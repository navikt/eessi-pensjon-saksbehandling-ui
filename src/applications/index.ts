import BUC from './BUC'
import Journalføring from './Journalføring'
import PageNotification from './PageNotification'
import S3Inventory from './S3Inventory'

export {
  BUC,
  Journalføring,
  PageNotification,
  S3Inventory
}
