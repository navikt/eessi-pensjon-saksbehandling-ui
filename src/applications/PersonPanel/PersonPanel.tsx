import { Accordion, Panel } from '@navikt/ds-react'
import {getPersonAvdodInfo, getPersonInfo} from 'src/actions/person'
import { AllowedLocaleString, FeatureToggles, PesysContext } from 'src/declarations/app.d'
import { PersonPDL } from 'src/declarations/person'
import { PersonAvdods } from 'src/declarations/person.d'
import { State } from 'src/declarations/reducers'
import { timeDiffLogger } from 'src/metrics/loggers'
import { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import PersonBody from './PersonBody'
import PersonTitle from './PersonTitle'
import {GJENNY, VEDTAKSKONTEKST} from "src/constants/constants";

export interface PersonPanelSelector {
  aktoerId: string | null | undefined
  avdodAktoerId: string | null | undefined
  featureToggles: FeatureToggles
  gettingPersonInfo: boolean
  locale: AllowedLocaleString
  personPdl?: PersonPDL,
  personAvdods: PersonAvdods | undefined,
  pesysContext: PesysContext | undefined,
  vedtakId: string | null | undefined
}

const mapState = (state: State): PersonPanelSelector => ({
  /* istanbul ignore next */
  aktoerId: state.app.params.aktoerId,
  avdodAktoerId: state.app.params.avdodAktoerId,
  featureToggles: state.app.featureToggles,
  gettingPersonInfo: state.loading.gettingPersonInfo,
  locale: state.ui.locale,
  personPdl: state.person.personPdl,
  personAvdods: state.person.personAvdods,
  pesysContext: state.app.pesysContext,
  vedtakId: state.app.params.vedtakId
})

export const PersonPanel = (): JSX.Element => {
  const { aktoerId, featureToggles, gettingPersonInfo, locale, personPdl, personAvdods, pesysContext, vedtakId }: PersonPanelSelector =
    useSelector<State, PersonPanelSelector>(mapState)
  const [totalTimeWithMouseOver, setTotalTimeWithMouseOver] = useState<number>(0)
  const [mouseEnterDate, setMouseEnterDate] = useState<Date | undefined>(undefined)

  useEffect(() => {
    return () => {
      timeDiffLogger('PersonPanel.mouseover', totalTimeWithMouseOver)
    }
  }, [])

  const dispatch = useDispatch()

  const onMouseEnter = () => setMouseEnterDate(new Date())

  const onMouseLeave = () => {
    if (mouseEnterDate) {
      setTotalTimeWithMouseOver(totalTimeWithMouseOver + (new Date().getTime() - mouseEnterDate?.getTime()))
    }
  }

  useEffect(() => {
    if (aktoerId && pesysContext && pesysContext !== GJENNY) {
        dispatch(getPersonInfo(aktoerId))
        if (pesysContext === VEDTAKSKONTEKST && !!vedtakId) {
          dispatch(getPersonAvdodInfo(aktoerId, vedtakId, featureToggles.NR_AVDOD as number | undefined))
        }
    }
  }, [aktoerId, pesysContext])

  return (
    <Panel
      border style={{ padding: '0px' }}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      <Accordion style={{ borderRadius: '4px' }} data-testid='w-PersonPanel-id'>
        <Accordion.Item>
          <Accordion.Header>
            <PersonTitle
              gettingPersonInfo={gettingPersonInfo}
              person={personPdl}
            />
          </Accordion.Header>
          <Accordion.Content>
            <PersonBody
              locale={locale}
              person={personPdl}
              personAvdods={personAvdods}
            />
          </Accordion.Content>
        </Accordion.Item>
      </Accordion>
    </Panel>
  )
}

export default PersonPanel
