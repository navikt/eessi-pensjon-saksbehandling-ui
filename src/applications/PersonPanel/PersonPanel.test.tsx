import { render, screen } from '@testing-library/react'
import { getPersonAvdodInfo, getPersonInfo } from 'src/actions/person'
import { BRUKERKONTEKST, VEDTAKSKONTEKST } from 'src/constants/constants'

import mockFeatureToggles from 'src/mocks/app/featureToggles'
import personAvdod from 'src/mocks/person/personAvdod'
import { stageSelector } from 'src/setupTests'
import { PersonPanel, PersonPanelSelector } from './PersonPanel'

jest.mock('src/actions/person', () => ({
  getPersonInfo: jest.fn(),
  getPersonAvdodInfo: jest.fn()
}))

describe('src/applications/PersonPanel/PersonPanel', () => {
  let wrapper: any

  const defaultSelector: PersonPanelSelector = {
    aktoerId: '123',
    avdodAktoerId: undefined,
    featureToggles: {
      ...mockFeatureToggles,
      NR_AVDOD: 1
    },
    gettingPersonInfo: false,
    locale: 'nb',
    personPdl: undefined,
    personAvdods: personAvdod(1),
    pesysContext: VEDTAKSKONTEKST,
    vedtakId: '345'
  }
  beforeEach(() => {
    stageSelector(defaultSelector, {})
    wrapper = render(<PersonPanel />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('UseEffect: fetches person info when mounting: VEDTAKSKONTEKST', () => {
    expect(getPersonAvdodInfo).toHaveBeenCalled()
  })

  it('UseEffect: fetches person info when mounting: not VEDTAKSKONTEKST', () => {
    (getPersonAvdodInfo as jest.Mock).mockReset();
    (getPersonInfo as jest.Mock).mockReset()
    stageSelector(defaultSelector, { pesysContext: BRUKERKONTEKST })
    wrapper = render(<PersonPanel />)
    expect(getPersonAvdodInfo).not.toHaveBeenCalled()
    expect(getPersonInfo).toHaveBeenCalled()
  })

  it('Render: has proper HTML structure', () => {
    expect(screen.getByTestId('w-PersonPanel-id')).toBeTruthy()
  })
})
