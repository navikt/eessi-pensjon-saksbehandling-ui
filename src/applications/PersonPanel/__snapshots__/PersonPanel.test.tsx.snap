// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/PersonPanel/PersonPanel Render: match snapshot 1`] = `
.c5 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g20[id="c4"] {
  content: "c5,";
}

.c3 {
  min-width: 40px;
  min-height: 40px;
  background-color: whitesmoke;
  border-radius: 100px;
  -webkit-transition: width .6s ease;
  transition: width .6s ease;
  -webkit-animation: da-daAO 1s linear infinite;
  animation: da-daAO 1s linear infinite;
  background-image: linear-gradient(45deg,hsla(90,90%,90%,.15) 25%,transparent 0,transparent 50%,hsla(90,90%,90%,.15) 0,hsla(90,90%,90%,.15) 75%,transparent 0,transparent);
}

ui-fss.g50[id="c2"] {
  content: "c3,";
}

.c7 {
  min-width: 50%;
  -webkit-transition: width .6s ease;
  transition: width .6s ease;
  background-color: whitesmoke;
  -webkit-animation: da-daAO 1s linear infinite;
  animation: da-daAO 1s linear infinite;
  background-image: linear-gradient(45deg,hsla(90,90%,90%,.15) 25%,transparent 0,transparent 50%,hsla(90,90%,90%,.15) 0,hsla(90,90%,90%,.15) 75%,transparent 0,transparent);
}

ui-fss.g51[id="c6"] {
  content: "c7,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

ui-fss.g52[id="c0"] {
  content: "c1,";
}

.c9 {
  height: 40px;
  min-height: 40px;
  width: 100%;
}

ui-fss.g53[id="c8"] {
  content: "c9,";
}

<div
  class="navds-panel navds-panel--border"
  style="padding: 0px;"
>
  <div
    class="navds-accordion"
    data-testid="w-PersonPanel-id"
    style="border-radius: 4px;"
  >
    <div
      class="navds-accordion__item"
    >
      <button
        aria-expanded="false"
        class="navds-accordion__header"
        type="button"
      >
        <span
          class="navds-accordion__header-content navds-heading navds-heading--small"
        >
          <div
            class="c0 c1"
          >
            <div
              class="c2 c3"
            />
            <div
              class="c4 c5"
            />
            <div
              class="c6 c7 c8 c9"
            />
          </div>
        </span>
        <svg
          aria-hidden="true"
          class="navds-accordion__expand-icon"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 17.414 4.293 9.707l1.414-1.414L12 14.586l6.293-6.293 1.414 1.414L12 17.414Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="navds-accordion__expand-icon navds-accordion__expand-icon--filled"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="m12 18.121-8.06-8.06 2.12-2.122L12 13.88l5.94-5.94 2.12 2.122L12 18.12Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        style="height: 0px; overflow: hidden;"
      >
        <div
          aria-hidden="true"
          style="display: none;"
        >
          <div
            class="navds-accordion__content navds-body-long"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
